### PR TITLE
Delete comment

### PR DESCRIPTION
**Pull request by Szpytma**

### CODE REVIEW

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
looking great so far! I just have a couple of things I'd like you to take a look at

- at the moment any user (they don't even have to be logged in) can delete any comment. Think about how similar websites would work and make it so that only the logged in user can delete their own comments (hardcoding a logged in user is fine)
- if the comment is or isn't deleted there isn't any feedback to the user at the moment. You might also want to consider something like confirming that the user wants to delete this comment before making the api call
- on slow 3g the comment isn't deleted straight away and the button is still enabled for the user to try and delete it multiple times before the api call is resolved

After those things are sorted send us another PR :)

File: src/components/CommentCard.jsx [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
     const date = new Date(dataString);
     const formattedDate = date.toLocaleString();
     return formattedDate;
   };
+  const [disabled, setDisabled] = useState(false);
+  const [removed, setRemoved] = useState(false);
+  const removeCommentByID = () => {
+    setDisabled(true);
+    setRemoved(true);
+    api.removeCommentByID(comment.comment_id).then(() => {
```
Comment (lewispricey) [inline src/components/CommentCard.jsx]: I really like this workaround to delete the comment optimistically 👍

File: src/components/CommentCard.jsx [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
+
   return (
     <div className="">
+      {removed && <p>comment removed</p>}
       <h3>{comment.body}</h3>
       <p>
         Posted by {comment.author} on {formatDate(comment.created_at)}
       </p>
       <p>❤️{comment.votes}</p>
+      {loggedUser === comment.author && (
```
Comment (lewispricey) [inline src/components/CommentCard.jsx:39]: Awesome way of just showing the delete button for only the user that origionaly posted it 🙌

**Review by lewispricey — APPROVED** [review]
Great work on this ticket Pawel thanks for implementing Roisin's feedback, I really like how you've gone about the login and only displaying the delete button on comments written by the logged in user!

The only thing I'd recommend adding is some sort of message if the delete failed (the user lost connection or a server error happened) 

There's also an extra "hello again from https://nc-games-jr4p.onrender.com/reviews/14" rendering at the top of the comments, I'm not sure where this is coming from but I'll send over a screenshot.

Overall, awsome work! Please could you add in the above then feel free to merge away 🚀

**Pull request merged**